DisguiseUnityRenderStream/Runtime/PipelineAbstraction/CameraCaptureDescription.cs

```diff
@@ -0,0 +1,213 @@
+
+        /// <summary>
+        /// When true, the final capture has flipped Y UV coordinates relative to <see cref="Camera.targetTexture"/>
+        /// depending on <see cref="m_autoFlipY"/> and the current graphics API.
+        /// </summary>
+        public bool NeedsFlipY => m_autoFlipY && SystemInfo.graphicsUVStartsAtTop;
+
+        /// <summary>
+        /// We leverage the hardware linear <-> sRGB automatic conversion when possible.
+        /// No hardware sRGB texture format is available for float textures for example.
+        /// </summary>
+        public bool NeedsSoftwareSRGBConversion => m_colorSpace == ColorSpace.sRGB && !GraphicsFormatUtility.IsSRGBFormat(m_colorFormat);
+        
+        public bool CameraTextureIsMSAA => m_msaaSamples > 1;
+
+        /// <summary>
+        /// Describes the texture to use for <see cref="Camera.targetTexture"/>.
+        /// </summary>
+        public RenderTextureDescriptor GetCameraDescriptor()
+        {
+            var descriptor = new RenderTextureDescriptor(m_width, m_height, m_colorFormat, m_depthBufferBits, 1);
+            descriptor.msaaSamples = m_msaaSamples;
+            descriptor.sRGB = m_colorSpace == ColorSpace.sRGB && !NeedsSoftwareSRGBConversion;
+            
+            return descriptor;
+        }
+        
+        /// <summary>
+        /// Describes the texture to contain the flipped and/or color space converted version of the <see cref="Camera.targetTexture"/>.
+        /// </summary>
+        public RenderTextureDescriptor GetCameraBlitDescriptor()
+        {
+            var descriptor = GetCameraDescriptor();
+            descriptor.depthBufferBits = 0;
+            descriptor.msaaSamples = 1;
+            
+            return descriptor;
+        }
+
+        /// <summary>
+        /// Describes the texture to use for storing the depth capture.
+        /// </summary>
+        public RenderTextureDescriptor GetDepthCopyDescriptor()
+        {
+            return new RenderTextureDescriptor(m_width, m_height, m_depthCopyFormat, 0, 1);
+        }
+        
+        public override bool Equals(object obj)
+        {
+            return obj is CameraCaptureDescription other && Equals(other);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = (int)m_colorSpace;
+                hashCode = (hashCode * 397) ^ (m_autoFlipY ? 1 : 0);
+                hashCode = (hashCode * 397) ^ m_width;
+                hashCode = (hashCode * 397) ^ m_height;
+                hashCode = (hashCode * 397) ^ (int)m_colorFormat;
+                hashCode = (hashCode * 397) ^ m_msaaSamples;
+                hashCode = (hashCode * 397) ^ m_depthBufferBits;
+                hashCode = (hashCode * 397) ^ (m_copyDepth ? 1 : 0);
+                hashCode = (hashCode * 397) ^ (int)m_depthCopyFormat;
+                hashCode = (hashCode * 397) ^ (int)m_depthCopyMode;
+                return hashCode;
```

Comment:
Question: I guess this is faster than `HashCode.Combine` (https://learn.microsoft.com/en-us/dotnet/api/system.hashcode.combine?view=net-7.0)?

Reply:
I found a warning in the MS doc: `Do not store hash codes produced by [HashCode](https://learn.microsoft.com/en-us/dotnet/api/system.hashcode?view=net-7.0) in serialized structures, for example, on-disk.`

We don't serialize the hashcodes anywhere at the moment but it would be an annoying problem to track down if we ever did.

Which is a shame because I really like the API!

Reply:
I might have missed something but as far as I can tell, storing the result of GetHashCode is a quite risky move anyway as the default implementation is not deterministic either (kind of the equivalent of returning the address of the object as an integer).

Also, Object.GetHashCode documentation (https://learn.microsoft.com/en-us/dotnet/api/system.object.gethashcode?view=net-8.0) states:

> A hash code is intended for efficient insertion and lookup in collections that are based on a hash table. A hash code is not a permanent value. For this reason:
> * Do not serialize hash code values or store them in databases.
> * ...